core/include/mbox.h

```diff
@@ -185,6 +185,17 @@ static inline size_t mbox_avail(mbox_t *mbox)
     return cib_avail(&mbox->cib);
 }
 
+/**
+ * @brief   Unset's the mbox, effectively deinitializing and invalidating it.
```

Comment:
From https://github.com/RIOT-OS/RIOT/pull/15484#discussion_r527685144

> This leaks a possibly supplied queue. Maybe better call it mbox_release_queue?
The mbox actually works without queue, it'd just be synchronous from that point on.
>
> This is also a bit dangerous, if this is called with messages in the queue, they get lost. If there's a sender that was previously waiting, it would be stuck there until the queue transitions from full to empty again.

The `sys_mbox_set_invalid()` function is called whenever a netconn (lwIP's sock equivalent) is closed or finished, so this isn't a risk, but maybe that function should be fixed in the way that readers and writers are also released and removed somehow.

Reply:
So I still see issues:
- as is, this is not thread safe, something could send/rx a message between those two lines. I think mbox breaks if the array is NULL but cib mask not. (this *could* be mitigated by changing the order, but that seems hacky).
- any waiters on that mbox would be blocked indefinitely.
   I don't see any nice way around this. even `while (mbox_try_ receive(tmp) {}` would be racey.

I'm fine adding this function (if needed), but maybe those should be fixed or mentioned?